woocommerce/woocommerce
# Extract the scanning functionality from code-analyzer, move common code into a separate package

### All Submissions:






### Changes proposed in this Pull Request:

1. In the release post generator we were calling the Analyzer command directly from CLI, so it was a black box. We couldn't reuse the functionality or customize it.

2. The analyzer was hacked a bit (by me) to support 2 use cases before, 1 was linting and the other producing output of changes for other tools to use. This extracts out the scanning code so it can be reused in both those contexts.

3. There was an emerging set of utilities being used in multiple tools, I've moved them into a `cli-core` package.

4. Removed use of oclif from analyzer and port to Commander.js (haven't removed oclif just yet as one other command uses it).






### How to test the changes in this Pull Request:

1. Try running the release post generator to make sure it still works. e.g. from `tools/release-post-generator`: `pnpm run release -- "6.8.0" --outputOnly`
2. Try running the new analyzer. Example command from `tools/code-analyzer`: `pnpm run analyzer  -- "6.8.0" "6.8.0"`
3. See the lint changes GH action should pass in this PR.

### Other information:




### FOR PR REVIEWER ONLY:
